### PR TITLE
[feature fix] Activate markup-safe rendering in profile pages (markup-2)

### PR DESCRIPTION
**Pull request by abought**
This PR reopens #2925, which was accidentally closed. Very sorry for the inconvenience.

Depends on #2917 and draws from #1308, #2933.

## Purpose
Turn on markup-safe rendering in the profile and settings pages.

See testing notes for route list and examples: https://github.com/CenterForOpenScience/osf.io/pull/3635#issuecomment-121980657

## Future work
There is an implied database migration for old entries in `Node` and `NodeLog`, as some old records may have been markup-escaped before saving to the DB. See https://github.com/CenterForOpenScience/osf.io/blob/30d3b46a2f0030672378ce84b5cc7accae3cb7ad/scripts/consistency/fix_html_entities.py

Some old records may show up incorrectly in the pages where markup-escaping is turned on, until the DB conversion is run (which would be done after all routes were converted).

### CODE REVIEW

**Comment by abought** [comment]
## Testing notes for QA (copied from #2925)
### Purpose of this PR
This PR is designed to make sure that text you type into the OSF is displayed exactly as you entered it (rather than being treated like code that changes how the OSF works).

Most HTML gets scrubbed out (you can't name a project `<i>QA is awesome!</i>` and have the name show up in italics), but some words display things oddly. For example, on project pages, naming something `Sub-component A &copy;` shows up differently than what you typed in:
screen shot 2015-07-15 at 1 51 34 pm

### Incremental!
We realize that changing how things are displayed is a headache to QA, so we are doing this in small chunks- a few pages at a time. This PR covers Profile and Settings pages.

### List of routes affected
All routes are prefixed with `osf.io` as appropriate. `<brackets>` indicate placeholders for the specified value.
- `/profile/`
- `/profile/<user_id>/`
- `/user/merge/` (Not sure if this page is ever used!)
- `/user/<user_id>/<project_id>/claim/` Where people go when asked to "claim" an email address that was added to a project.
- `/user/<uid>/<pid>/claim/verify/<token>/` View that prompts user to enter their password in order to claim contributorship on a project.
- `/settings` 
- `/settings/account/`
- `/settings/addons/`
- `/settings/notifications/`

### What to look out for
Try a variety of fields that display information you entered in. (think profile fields, schools, URLs, etc) This should not change any validation or entry rules- only how what you entered is displayed.

- JavaScript console errors
- Weird sequences of code where you would expect the addon configuration box to go
- Projects (with 1, 2, or n authors) that render weirdly on the profile page "list of public projects"
- Text that isn't displayed exactly as you type it in- including special characters (see below), or unicode

### References
- List of HTML escape characters (`&amp;`, `&gt;`, and `&lt;` are probably sufficient for testing!) http://www.w3.org/html/wg/drafts/html/master/syntax.html#named-character-references
- Unicode emoticons: http://unicodeemoticons.com/
- QA probably has a list of common test strings- try those! I sometimes use this one
`☃Backslash'\d project & name &copy; alert('hello, world!') ¿ " more \\ \backonechar Text > <wbr> &lt;i&gt;Italicized&lt;/i&gt;☃`

**Comment by GaryKriebel** [comment]
&<> and &#169;
screen shot 2015-07-16 at 12 49 31 pm

 will become &amp;&lt;&gt; and © in the form fields and general display
screen shot 2015-07-16 at 12 49 53 pm

screen shot 2015-07-16 at 12 50 53 pm

but will be rendered as the original characters in the nav bar

screen shot 2015-07-16 at 12 50 58 pm

**Comment by GaryKriebel** [comment]
The behaviour noted above also happens on staging.

**Comment by abought** [comment]
These fields are filled in by Knockout.js, and they are a faithful representation of how data looks in the database.

Typically we deal with this using the `safe_unescape_html` function on the JSON endpoints used to serve the data. Is converting that (and QAing every use of those apiV1 routes) in scope for this PR?

**Comment by lbanner** [comment]
Probably not, as long as it is noted. Yes, separate issue/PR for the fix is a good plan.

File: website/project/views/contributor.py [hunk 1]
```diff
@@ -551,7 +551,7 @@ def claim_user_registered(auth, node, **kwargs):
         logout_url = web_url_for('auth_logout', redirect_url=request.url)
         data = {
             'message_short': 'Already a contributor',
-            'message_long': ('The logged-in user is already a contributor to this'
+            'message_long': ('The logged-in user is already a contributor to this '
```
Comment (lyndsysimon) [inline website/project/views/contributor.py]: :baseball: Good catch.

File: website/routes.py [hunk 2]
```diff
@@ -81,7 +81,7 @@ def get_globals():
 class OsfWebRenderer(WebRenderer):
     """Render a Mako template with OSF context vars.
 
-    :param trust: Optional. If ``False``, markup-save escaping will be enabled
+    :param trust: Optional. If ``False``, markup-safe escaping will be enabled
```
Comment (lyndsysimon) [inline website/routes.py]: :baseball:

File: website/templates/profile.mako [hunk 3]
```diff
@@ -150,7 +149,7 @@
                    "tpl" : "util/render_nodes.mako",
                    "uri" : "/api/v1/profile/${profile["id"]}/public_projects/",
                    "replace" : true,
-                   "kwargs" : {"sortable" : true, "user": ${json.dumps(user)}, "pluralized_node_type": "projects", "skipBindings": true}
+                   "kwargs" : {"sortable" : true, "user": ${ user | sjson, n }, "pluralized_node_type": "projects", "skipBindings": true}
```
Comment (lyndsysimon) [inline website/templates/profile.mako]: I'm still not crazy about calling this `sjson` in templates. `to_json` would be my preference, followed by `j`.

I can live with it as-is, though.

**Comment by lyndsysimon** [comment]
First pass complete - no changes required.

**Comment by abought** [comment]
Updated PR and fixed one bug in rendering of profile node list (for projects with private parents).

File: website/util/sanitize.py [hunk 4]
```diff
@@ -14,6 +14,7 @@ def strip_html(unclean):
     return bleach.clean(unclean, strip=True, tags=[], attributes=[], styles=[])
 
 
+# TODO: Not used anywhere except unit tests? Review for deletion
```
Comment (sloria) [inline website/util/sanitize.py]: Good catch. I believe this function can be removed, but that can be in a separate PR.
